### ISSUE TITLE
Ghosting activity - Track node lifecycle activity from parent node perspective

### ISSUE THREAD

**Issue by scd** [theCrag/website#1972]
We need to be able to track the node lifecycle for restucture events. We should never forget where the route was originally created. We are currently doing a good job at keeping track of a route from the perspective of a route, but not such a good job a tracking how an area's child nodes are changing.

1. Create a route, rename it then later delete it then activity flow should be
activity reporting on the parent
  * new child Rx registered
  * rename Rx to Ry
  * deleted Ry
Note that this means we have to store the historical names.

2. Create a route R and reparent from P1 to P2
activity reporting on the route R
 * register R
 * reparented R from P1 to P2
activity reporting on the original parent node P1
 * new child R registered
 * moved R to P2
activity reporting on the new parent node P2
 * register R (rolled up from R)
 * reparent R from P1 to P2 (rolled up from R)
 * new child R due to reparent
Note that the activity associated with R itself will be rolled up to the new parent P1 due to the activity log facet. When R is moved away from the old parent the there needs to be two remnant activity records that stay with the old parent so that it is clear that the route was originally created there then moved out). 
Finally there needs to be a record in the new parent that stays with the new parent in case the route is moved again.

3.  Create Route R1 in A1, merge with R2 in A2
activity reporting on A1
  * new child R1 registered
  * removed R1 due to merge with A1
activity reporting on R2
  * R1 merged to R2

The enhancement is creating second restructure activity logs for registering/renaming/reparenting/deleting/merging nodes which stays with the parent.

In otherwords when you create a route you are creating two activity logs - one for the route and one for the parent. The routes activity log will follow the route as it undergoes restructures, while the parent logs remain unchanged.

Child Restructure
  * New registered child
  * New reparented child
  * Renamed child
  * Removed deleted child
  * Removed reparented child
  * Removed merged child

I need to tackle this now, because there is a dependency in the streams. If we want to include merge and reparent in the streams we need to know what node to aggregate on, which could involve two different crags. If we have the second activity log then this works great in the streams. In otherwords reparenting a cliff from one crag to another may create a stream event in both crags.

**Comment by scd** [comment]
I have renamed this issue to ghosting activity. There is some activity items which change a parents list of children which may later be lost if that child is archived, deleted, merged or moved. 

The activity which effects a parents children in this way is:
  * registering a route/area
  * deleting a route/area
  * archiving/unarchiving a route
  * merging a route/area
  * deleting a route/area
  * renaming a route/area

For each of these activities I have created a second ghost activity record which links to the primary activity record. The ghost record is connected to the parent record so it will not disappear if the route is moved/deleted/merged. 

The first structural problem I encountered with this was that the activity log was showing confusing dups, which was particularily annoying when you create mulitple routes. I now only show the ghost activity if the primary activity is no longer associated with parent record.

This means if you move a route to another area then the ghost activity log item will activated.

I am pretty stoked about this because it now means we have an audit trail in our activity log. Also it is was a prerequisite for working out how to handle reparenting and merging events in the stream events.
